lib/plugins/init/index.js

```diff
@@ -0,0 +1,118 @@
+'use strict';
+
+const BbPromise = require('bluebird');
+const path = require('path');
+
+class Init {
+  constructor(serverless, options) {
+    this.serverless = serverless;
+    this.options = options;
+
+    this.commands = {
+      init: {
+        usage: 'Initializes a provider with the Serverless Framework',
+        lifecycleEvents: [
+          'init',
+        ],
+        options: {
+          provider: {
+            usage: 'Name of the provider',
+            required: true,
+            shortcut: 'p',
+          },
+          key: {
+            usage: 'Access key for the provider',
+            shortcut: 'k',
+          },
+          secret: {
+            usage: 'Secret key for the provider',
+            shortcut: 's',
+          },
+          profile: {
+            usage: 'Name of the profile you wish to create. Defaults to \'default\'',
+            shortcut: 'n',
+          },
+        },
+      },
+    };
+
+    this.hooks = {
+      'init:init': () => BbPromise.bind(this)
+        .then(this.init),
+    };
+  }
+
+  init() {
+    // Sanitize
+    this.options.provider = this.options.provider.toLowerCase();
+    this.options.profile = this.options.profile ? this.options.profile : 'default';
+
+    // Validate
+    if (['aws'].indexOf(this.options.provider) < 0) {
+      throw new this.serverless.classes.Error('Only \'aws\' is supported at this time.');
+    }
+
+    // Init AWS
+    return this.initAws();
+  }
+
+  /**
+   * Init AWS
+   * - Saves AWS API Keys to a profile on the file system
+   */
+
+  initAws() {
+    // Validate
+    if (!this.options.key || !this.options.secret) {
+      throw new this.serverless.classes.Error('Please include --key and --secret options for AWS.');
+    }
+
+    // Inform
+    this.serverless.cli.log('Initializing AWS...');
+    this.serverless.cli.log('Saving your AWS profile in \'~/.aws/credentials\'...');
+
+    // Locate home directory on user's machine
+    const env = process.env;
+    const home = env.HOME ||
+      env.USERPROFILE ||
+      (env.HOMEPATH ? ((env.HOMEDRIVE || 'C:/') + env.HOMEPATH) : null);
+
+    if (!home) {
+      throw new this.serverless.classes.Error('Can\'t find home directory ' +
+        'on your local file system.');
+    }
+
+    // Check if ~/.aws/credentials exists
+    const configDir = path.join(home, '.aws');
+    const credsPath = path.join(configDir, 'credentials');
+
+    if (this.serverless.utils.fileExistsSync(credsPath)) {
+      // Check if credentials files contains anything
+      const credsFile = this.serverless.utils.readFileSync(credsPath);
+
+      // If credentials file exists w/ profile, exit
+      if (credsFile.length
+        && credsFile.indexOf(`[${this.options.profile}]`) > -1) {
```

Comment:
@ac360 thx for this part! was quite helpful - especially the home directory finding.

btw in case you every need it here a regex, that actually checks if the keys are set:

```
const credentialsRegexp = new RegExp(`\\[${escapeRegExp(profile)}\\]\naws_access_key_id=.+\naws_secret_access_key=.+\n`, 'g')
```